### ISSUE TITLE
Redesign pause screen UI - ideas here

### ISSUE THREAD

**Issue by hrydgard** [hrydgard/ppsspp#13016]
Currently it's way too easy to misclick, because to handle 5 save states on-screen we've had to make the save/load state buttons too small.

Floe proposed the following UI (clickable mockup: https://app.moqups.com/NKxxdPt3AY/view/page/ad64222d5 ) where you first click whether to save or load, then click the state.

There are many other possibilities though, like first click the slot, then save/load, for example. Not sure what's safest against misclicks. Maybe even easy access to an Undo Save State function would be a reasonable workaround...

**Comment by iota97** [comment]
IMHO it's better to have save/load button and only after the slot selection: 

if one miss-click can still cancel any action, also this will have 2 button on main screen and leave the 5 to the bigger popup. It's possible to select the wrong slot but the worst case is to overwrite the wrong one (if a user want to keep one safe could still probably use like the 5th slot and only the first for other making them further away), still way better than to load when you want to save or viceversa.

**Comment by LunaMoo** [comment]
Personally never liked the Save State/Load State repetition on the pause screen, but I don't think it needs a mayor redesign like proposed above. I recall we already had an issue about the missclick on android problem and recall my idea was to move it out to savestate preview screen.

 Save and load could simply be removed and keep the Save and Load on the bottom of image preview while Undo on the upper right of it. "Back" could be removed from preview dialog and just clicking on the image or anywhere outside of the preview window would leave the dialog without having to choose the option. An empty slot could have a "save" in place of the miniature to point the users where they should be clicking.

**Comment by unknownbrackets** [comment]
Right, as LunaMoo mentioned - this is exactly the same as #7437 and we should keep just one.

IMHO, it's nice seeing which save states are available at a glance.  Personally, I'd prefer the confirmation/buttons inline.  For example, first it shows:

```
AAAAAAAAA  2020-11-30  23:33:11
AAA 1 AAA  
AAAAAAAAA  Saved 3 hours ago
```

If you tap the image, the same thing that happens now would happen.  If you tap the rest, it'd slide in:

```
AAAAAAAAA  
AAA 1 AAA  [ Load State ]    [ Save State ]
AAAAAAAAA  
```

But we could also keep the buttons and show a confirmation in the same way.

While thinking about this UI, it'd be nice to introduce an "Undo" button for saves made recently.  We could even save state to a temp slot before loading, and allow an undo.  I think that would fully address the misclick problem.

That said, it can just be in the dialog and I think that'd be okay too.

-[Unknown]

**Comment by Gamemulatorer** [comment]
This one looks better imoo :)
> Some people said it may get pressing wrong button.
> ![01](https://cloud.githubusercontent.com/assets/3481559/6057944/a6056508-ad5d-11e4-914b-ca790427ece1.png)
> 
> better change to
> ![02](https://cloud.githubusercontent.com/assets/3481559/6057951/b0faea0a-ad5d-11e4-8f4e-f8d034b97f5e.png)

**Comment by SeventhSentinel** [comment]
I would appreciate the addition of a button that exits to desktop. In one of my setups,the most convenient way to close the emulator is by exiting to the menu, then exiting to desktop, and I'd like it if I could close PPSSPP in one menu instead of two.

**Comment by hrydgard** [comment]
One other thing I want to do is a "transparent" mode where you can edit post processing settings and aspect ratio etc and immediately see the effects, on the paused game - the controls would hover over the screen. I'm imagining an "Display settings" button on the page screen that switches it to such a mode.

**Comment by hrydgard** [comment]
We now have transparent background, and the post processing stuff is handled in separate issues. Moving further improvements here to 1.15.

**Comment by leoxxx** [comment]
> We now have transparent background, and the post processing stuff is handled in separate issues. Moving further improvements here to 1.15.

https://github.com/hrydgard/ppsspp/issues/16593
I don't want use transparent background, how can I use old one?

**Comment by hrydgard** [comment]
@leoxxx I'm adding an option in #16595

**Comment by YAETHeryk** [comment]
https://github.com/hrydgard/ppsspp/pull/16398
Although the horizontal layout can prevent misclicking, a new problem arises. The loading button is farther away from the edge of the screen. If you use a mobile phone to play, this button is not as easy to reach with your thumb (as before).

**Comment by hrydgard** [comment]
Yeah, true. Maybe should have put them in the other order? Not sure what people do the most often. If you have ideas, do post them.

**Comment by Gamemulatorer** [comment]
Maybe add an additional option to switch position "Savestate" and "Loadstate"?

**Comment by YAETHeryk** [comment]
Or maybe move the small save image to the center and the button to the side, I'm not sure if it will look ugly
